### PR TITLE
[SB] Catch websocket exception

### LINKED ISSUE #34300: Uncaught WebSocketConnectionClosedException
- **Package Name**: azure.servicebus
- **Package Version**: 7.11.4
- **Operating System**: MacOS
- **Python Version**: Python 3.9.18 [Clang 15.0.0 (clang-1500.1.0.2.5)] on darwin

**Describe the bug**

The following exception comes up when receiving messages from a client created with the `transport_type=azure.servicebus.TransportType.AmqpOverWebsocket` option.

```
Unexpected error occurred (WebSocketConnectionClosedException('socket is already closed.')). Handler shutting down.
Traceback (most recent call last):
  File "<REDACTED>/site-packages/azure/servicebus/_base_handler.py", line 421, in _do_retryable_operation
    return operation(**kwargs)
  File "<REDACTED>/site-packages/azure/servicebus/_servicebus_receiver.py", line 443, in _receive
    receiving = amqp_receive_client.do_work()
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/client.py", line 428, in do_work
    return self._client_run(**kwargs)
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/client.py", line 856, in _client_run
    self._connection.listen(wait=self._socket_timeout, **kwargs)
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/_connection.py", line 777, in listen
    if self._read_frame(wait=wait, **kwargs):
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/_connection.py", line 266, in _read_frame
    new_frame = self._transport.receive_frame(**kwargs)
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/_transport.py", line 475, in receive_frame
    header, channel, payload = self.read(**kwargs)
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/_transport.py", line 413, in read
    read_frame_buffer.write(read(8, buffer=frame_header, initial=True))
  File "<REDACTED>/site-packages/azure/servicebus/_pyamqp/_transport.py", line 779, in _read
    data = self.sock.recv()
  File "<REDACTED>/site-packages/websocket/_core.py", line 388, in recv
    opcode, data = self.recv_data()
  File "<REDACTED>/site-packages/websocket/_core.py", line 416, in recv_data
    opcode, frame = self.recv_data_frame(control_frame)
  File "<REDACTED>/site-packages/websocket/_core.py", line 437, in recv_data_frame
    frame = self.recv_frame()
  File "<REDACTED>/site-packages/websocket/_core.py", line 478, in recv_frame
    return self.frame_buffer.recv_frame()
  File "<REDACTED>/site-packages/websocket/_abnf.py", line 363, in recv_frame
    self.recv_header()
  File "<REDACTED>/site-packages/websocket/_abnf.py", line 319, in recv_header
    header = self.recv_strict(2)
  File "<REDACTED>/site-packages/websocket/_abnf.py", line 398, in recv_strict
    bytes_ = self.recv(min(16384, shortage))
  File "<REDACTED>/site-packages/websocket/_core.py", line 563, in _recv
    return recv(self.sock, bufsize)
  File "<REDACTED>/site-packages/websocket/_socket.py", line 90, in recv
    raise WebSocketConnectionClosedException("socket is already closed.")
websocket._exceptions.WebSocketConnectionClosedException: socket is already closed.
```

**To Reproduce**

Steps to reproduce the behaviour:

1. Create a client with `transport_type=azs.TransportType.AmqpOverWebsocket`
2. Receive messages form it for some time. I'm using this kind of setup based on the example in the SDK:

```
for _ in range(3):
    try:
        # Receive a message from the queue one at a time.
        messages = receiver.receive_messages(max_message_count=1, max_wait_time=5)
        break
    except azure.servicebus.exceptions.ServiceBusError as e:
        # Any other undefined service errors during settlement.
        # Can be transient, and can retry, but should be logged, and alerted on high volume.
        print("ServiceBusError during message retrieval. Retrying.")
        continue
```

And it does successfully recover from the `WebSocketConnectionClosedException`.

**Expected behavior**

I would expect the exception not to be printed, and instead just got the `ServiceBusError`, or whatever might be the appropriate exception from `azure.servicebus.exceptions`, but not one from `websocket._exceptions`.

**Additional context**

The error log seems to be emitted from [here](https://github.com/Azure/azure-sdk-for-python/blob/5a69366b7201461469879229ad517d24d94fafbb/sdk/servicebus/azure-servicebus/azure/servicebus/_transport/_pyamqp_transport.py#L1053).

### PR DESCRIPTION
**Pull request by l0lawrence**
Fixes #34300

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py [hunk 1]
```diff
@@ -794,6 +794,8 @@ def _read(self, n, initial=False, buffer=None, _errnos=None):  # pylint: disable
                 raise IOError("Websocket connection has already been closed.") from None
             except WebSocketTimeoutException as wte:
                 raise TimeoutError('Websocket receive timed out (%s)' % wte) from wte
+            except WebSocketConnectionClosedException as e:
```
Comment (l0lawrence) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py:797]:
```suggestion
            except (WebSocketConnectionClosedException, SSLError) as e:
```

Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py:797]: Would an import be needed for SSLError? Are we seeing this pop up anywhere?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py [hunk 2]
```diff
@@ -771,7 +771,7 @@ def _read(self, n, initial=False, buffer=None, _errnos=None):  # pylint: disable
         :return: The data read.
         :rtype: bytearray
         """
-        from websocket import WebSocketTimeoutException
+        from websocket import WebSocketTimeoutException, WebSocketConnectionClosedException
```
Comment (swathipil) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py:774]: can we add a test for this or is this only reproduceable manually?

Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_transport.py:774]: manually requires loss

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**